src/lib/attach-action-provider.ts

```diff
@@ -0,0 +1,56 @@
+import { Action, StateContext } from '@ngxs/store';
+import { ActionHandlerMetaData } from '@ngxs/store/src/actions/symbols';
+import { MetaDataModel } from '@ngxs/store/src/internal/internals';
+import * as lodash from 'lodash';
+
+const META_KEY = 'NGXS_META';
+const META_OPTIONS_KEY = 'NGXS_OPTIONS_META';
+
+export function attachActionProvider(storeClass: any, providers: any[]): void {
+  if (!storeClass[META_OPTIONS_KEY]) {
+    throw new Error('storeClass is not a valid NGXS Store');
+  }
+  providers.forEach(provider => {
+    const providedActions: ActionHandlerMetaData[] = parseProvidedActions(provider);
+    providedActions.forEach(providedAction => {
+      attachToStoreClass(storeClass, provider, providedAction);
+    });
+  });
+}
+
+const parseProvidedActions = (provider: any): ActionHandlerMetaData[] => {
+  const metaDataModel: MetaDataModel = provider[META_KEY];
+  if (!metaDataModel) {
+    throw new Error('provider is not a valid NGXS Action Provider');
+  }
+  const actions: ActionHandlerMetaData[][] = Object.values(metaDataModel.actions);
+  return lodash.flattenDeep(actions);
```

Comment:
to keep the dependencies law, could we remove loadash und use a native implementation fo flattenDeep like: 
```
const flattenDeep = (arr) => Array.isArray(arr)
  ? arr.reduce( (a, b) => a.concat(flattenDeep(b)) , [])
  : [arr]
```